packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -39,10 +55,35 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
     const signerType = loadSignerType();
     if (signerType) {
       this.signer = this.initSigner(signerType);
+      logSignerLoadedFromStorage({ signerType });
     }
   }
 
   public async request<T>(args: RequestArguments): Promise<T> {
+    // correlation id across the entire request lifecycle
+    const correlationId = crypto.randomUUID();
+    logRequestStarted({ method: args.method, correlationId });
+
+    try {
+      const result = await this._request(args, correlationId);
```

Comment:
would it not be better to store/retrieve the correlationId in the zustand store vs modifying function args to pass it down?

Reply:
+1 this seems like a better approach we can have a map between request id and the request that clear when the requests resolve

Reply:
switched to use a non persisted map of request object to ids in zustand!